system-addon/common/Reducers.jsm

```diff
@@ -91,7 +96,39 @@ function TopSites(prevState = INITIAL_STATE.TopSites, action) {
   }
 }
 
+function AutoMigrate(prevState = INITIAL_STATE.AutoMigrate, action) {
+  switch (action.type) {
+    case at.AUTOMIGRATE_HIDE: {
+      return {
+        display: false,
+        stage: 0,
+        msg: ""
+      };
+    }
+    case at.AUTOMIGRATE_MIGRATED: {
+      if (!action.data && !action.data.msg) {
+        return prevState;
+      }
+      let {msg} = action.data;
+      return {
+        display: true,
+        stage: 0,
+        msg
+      };
+    }
+    case at.AUTOMIGRATE_IS_REVERTED: {
+      return {
+        display: true,
+        stage: 1,
```

Comment:
@k88hudson not sure if you have preferences on how this `stage` / state should be handled. With just numbers as here? Or strings or enum-ish? Or I suppose completely different is just deriving both `display` and `stage` from the component and storing the `state` as "HIDDEN" "MIGRATED" "REVERTED" ?